### PR TITLE
Terminology Update: .NET runtime -> .NET implementation

### PR DESCRIPTION
**Pull request by tompratt-AQ**
## Summary
Of the 62 instances of ".NET runtime(s)", I changed 27 instances across 12 files to ".NET implementation(s)".

I didn't change 35 instances in 17 files for the following reasons:

1. The instance referred to "ASP.NET runtime"
2. The instance was the name of a native image service (for example, ".NET Runtime Optimization Service v2.0.50727_X86").
3. The instances were in "components.md", already completed in [PR 2753](https://github.com/dotnet/docs/pull/2753).

## Suggested Reviewers
@rpetrusha

### CODE REVIEW

File: docs/core/index.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ The following characteristics best define .NET Core:
 
 .NET Core is composed of the following parts:
 
-- A [.NET runtime](https://github.com/dotnet/coreclr), which provides a type system, assembly loading, a garbage collector, native interop and other basic services. 
+- A [.NET implementation](https://github.com/dotnet/coreclr), which provides a type system, assembly loading, a garbage collector, native interop and other basic services.
```
Comment (rpetrusha) [inline docs/core/index.md]: runtime, and not implementation, is correct here.

File: docs/core/docker/building-net-docker-images.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ Latest versions of each variant:
 - `microsoft/dotnet:core-deps`
 
 Here is a list of the images after a `docker pull <imagename>` on a development machine to show the various sizes. Notice, the development/build variant, `microsoft/dotnet:1.0.0-preview2-sdk` is larger as it contains the SDK to develop and build your application. The production variant, `microsoft/dotnet:core` is smaller, as it only contains the .NET Core runtime. 
-The minimal image capable of being used on Linux, `core-deps`, is quite smaller, however your application will need to copy a private copy of the .NET Runtime with it. Since containers are already private isolation barriers, you will lose that optimization when running multiple dotnet based containers. 
+The minimal image capable of being used on Linux, `core-deps`, is quite smaller, however your application will need to copy a private copy of the .NET implementation with it. Since containers are already private isolation barriers, you will lose that optimization when running multiple dotnet based containers.
```
Comment (rpetrusha) [inline docs/core/docker/building-net-docker-images.md]: runtime, and not implementation, is correct here.

File: docs/core/docker/building-net-docker-images.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ In addition to the optimized scenarios of development, build and production, we
...
+- `microsoft/dotnet:<version>-core-deps` : that is **microsoft/dotnet:1.0.0-core-deps**, if you wish to run self-contained applications use this image. It contains the operating system with all of the native dependencies needed by .NET Core. This image can also be used as a base image for your own custom CoreFX or CoreCLR builds. While the **onbuild** variant is optimized to simply place your code in an image and run it, this image is optimized to have only the operating system dependencies required to run .NET Core apps that have the .NET implementation packaged with the application. This image isn't generally optimized for running multiple .NET Core containers on the same host, as each image carries the .NET Core runtime within the application, and you will not benefit from image layering.
```
Comment (rpetrusha) [inline docs/core/docker/building-net-docker-images.md]: runtime, and not implementation, is correct here.

File: docs/standard/exceptions/best-practices-for-exceptions.md [hunk 4]
```diff
@@ -94,7 +94,7 @@ For an example, see [How to: Create User-Defined Exceptions](how-to-create-user-
...
+For example, on .NET implementations that implement App Domains, exceptions may occur across App domains. Suppose App Domain A creates App Domain B, which executes code that throws an exception. For App Domain A to properly catch and handle the exception, it must be able to find the assembly that contains the exception thrown by App Domain B. If App Domain B throws an exception that is contained in an assembly under its application base, but not under App Domain A's application base, App Domain A will not be able to find the exception, and the common language runtime will throw a @System.IO.FileNotFoundException exception. To avoid this situation, you can deploy the assembly that contains the exception information in two ways:
```
Comment (rpetrusha) [inline docs/standard/exceptions/best-practices-for-exceptions.md]: implement App Domains --> support App Domains

File: docs/standard/tour.md [hunk 5]
```diff
@@ -52,13 +52,13 @@ In the following example, the runtime throws an `InvalidIndexException` exceptio
 
 ## Working with unmanaged resources
 
-Some objects reference *unmanaged resources*. Unmanaged resources are resources that aren't automatically maintained by the .NET runtime. For example, a file handle is an unmanaged resource. A <xref:System.IO.FileStream> object is a managed object, but it references a file handle, which is unmanaged. When you're done using the <xref:System.IO.FileStream>, you need to release the file handle.
+Some objects reference *unmanaged resources*. Unmanaged resources are resources that aren't automatically maintained by the .NET implementation. For example, a file handle is an unmanaged resource. A <xref:System.IO.FileStream> object is a managed object, but it references a file handle, which is unmanaged. When you're done using the <xref:System.IO.FileStream>, you need to release the file handle.
```
Comment (rpetrusha) [inline docs/standard/tour.md]: runtime, rather than implementation, is correct here.

File: docs/standard/tour.md [hunk 6]
```diff
@@ -52,13 +52,13 @@ In the following example, the runtime throws an `InvalidIndexException` exceptio
...
 
 In .NET, objects that reference unmanaged resources implement the <xref:System.IDisposable> interface. When you're done using the object, you call the object's <xref:System.IDisposable.Dispose> method, which is responsible for releasing any unmanaged resources. .NET languages provide a convenient `using` syntax for such objects, as shown in the following example:
 
 [!code-csharp[UnmanagedResources](../../samples/csharp/snippets/tour/UnmanagedResources.csx#L1-L6)]
 
-Once the `using` block completes, the .NET runtime automatically calls the `stream` object's <xref:System.IDisposable.Dispose> method, which releases the file handle. The runtime also does this if an exception causes control to leave the block.
+Once the `using` block completes, the .NET implementation automatically calls the `stream` object's <xref:System.IDisposable.Dispose> method, which releases the file handle. The runtime also does this if an exception causes control to leave the block.
```
Comment (rpetrusha) [inline docs/standard/tour.md]: runtime, not implementation, is correct here.

File: docs/standard/tour.md [hunk 7]
```diff
@@ -70,7 +70,7 @@ For more details, see the following topics:
 
 An object is an instance of a specific type. The only operations allowed for a given object are those of its type. A `Dog` type may have `Jump` and `WagTail` methods but not a `SumTotal` method. A program only calls the methods belonging to a given type. All other calls result in either a compile-time error or a run-time exception (in case of using dynamic features or `object`).
 
-.NET languages are object-oriented with hierarchies of base and derived classes. The .NET runtime only allows object casts and calls that align with the object hierarchy. Remember that every type defined in any .NET language derives from the base <xref:System.Object> type.
+.NET languages are object-oriented with hierarchies of base and derived classes. The .NET implementation only allows object casts and calls that align with the object hierarchy. Remember that every type defined in any .NET language derives from the base <xref:System.Object> type.
```
Comment (rpetrusha) [inline docs/standard/tour.md]: runtime, not implementation, is correct here.

File: xml/System.Data.SqlClient/SqlCommand.xml [hunk 8]
```diff
@@ -888,7 +888,7 @@ class Program {
...
   
- Because the callback procedure executes from within a background thread supplied by the Microsoft .NET runtime, it is very important that you take a rigorous approach to handling cross-thread interactions from within your applications. For example, you must not interact with a form's contents from within your callback procedure; should you have to update the form, you must switch back to the form's thread in order to do your work. The example in this topic demonstrates this behavior.  
+ Because the callback procedure executes from within a background thread supplied by the Microsoft .NET implementation, it is very important that you take a rigorous approach to handling cross-thread interactions from within your applications. For example, you must not interact with a form's contents from within your callback procedure; should you have to update the form, you must switch back to the form's thread in order to do your work. The example in this topic demonstrates this behavior.
```
Comment (rpetrusha) [inline xml/System.Data.SqlClient/SqlCommand.xml]: runtime, not implementation, is correct here.

**Review by rpetrusha — CHANGES_REQUESTED** [review]
@tompratt-AQ, one suggested change, and a number of changes that should be reverted.

**Comment by tompratt-AQ** [comment]
@rpetrusha I made your review changes, but also changed instances of ".NET platform" to ".NET implementations" in the following files, so they will need review:
 
docs/core/index.md
docs/core/versions/index.md 
docs/standard/assembly-format.md 
docs/standard/net-standard.md 
docs/standard/tour.md   

The rest of the changes for ".NET platform" are in PR #2831.

**Comment by rpetrusha** [comment]
Your changes all look good, @tompratt-AQ. I'll merge now.

**Review by rpetrusha — APPROVED** [review]

**Pull request merged**